Content.Shared/Chat/V2/Moderation/SimpleCensor.cs

```diff
@@ -0,0 +1,272 @@
+﻿using System.Collections.Frozen;
+using System.Linq;
+using System.Text;
+using System.Text.Unicode;
+
+namespace Content.Shared.Chat.V2.Moderation;
+
+/// <summary>
+/// A basic censor. Mostly built from reference from Go-Away. Not bullet-proof.
+/// </summary>
+public sealed class SimpleCensor
+{
+    private bool _sanitizeSpecialCharacters;
+    private bool _sanitizeL33TSpeak;
+    private bool _sanitizeSpaces;
```

Comment:
This could code do with some better comments and naming. It wasn't clear to me when I started reading that these `_sanitize` parameters are for folding down inputs from common substitutions so they can be matched.